app/controllers/application_controller.rb

```diff
@@ -1,5 +1,27 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :mailbox, :conversation
+
+  def current_user
+    User.find_by(id: session[:user_id])
+  end
+
+  def require_login (redirect_path = root_path)
+    redirect_to redirect_path unless is_authenticated?
+  end
+
+  def is_authenticated?
+    !!session[:user_id]
+  end
+
+  private
+
+  def mailbox
+    @mailbox ||= current_user.mailbox
+  end
+
+  def conversation
+    @conversation ||= mailbox.conversations.find(params[:id])
```

Comment:
Depending on the controller this runs in, params[:id] might not be the right id.